pkg/server/image_pull.go

```diff
@@ -253,6 +256,36 @@ func (c *criService) getResolver(ctx context.Context, ref string, cred func(stri
 	if err != nil {
 		return nil, imagespec.Descriptor{}, errors.Wrap(err, "parse image reference")
 	}
+
+	httpClient := http.DefaultClient
+
+	if auth, ok := c.config.Registry.Auths[refspec.Hostname()]; ok {
```

Comment:
This is not the right place to do this.

For example, for image `docker.io/library/busybox:latest` with registry mirror `mirror.gcr.io`, the `refspec.Hostname()` here will be `docker.io`, which may not be the actual endpoint we are going to pull from `mirror.gcr.io`.

This should be done in the following loop.

Reply:
Moved it inside the loop as well. I assume if nothing in the loop matches we still need to handle the default case that is why I left it at https://github.com/containerd/cri/pull/1144/files#diff-87348f253685e1fb4a3238134a550238R319 as well. Let me know if there is a better way to handle this.